mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/mapping/EmbeddableIdMappingTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.test.mapping;
+
+import static org.hibernate.ogm.datastore.mongodb.utils.MongoDBTestHelper.assertDbObject;
+
+import java.util.Collections;
+
+import org.hibernate.Transaction;
+import org.hibernate.ogm.OgmSession;
+import org.hibernate.ogm.backendtck.id.Label;
+import org.hibernate.ogm.backendtck.id.News;
+import org.hibernate.ogm.backendtck.id.NewsID;
+import org.hibernate.ogm.backendtck.id.embeddable.SingleBoardComputer;
+import org.hibernate.ogm.backendtck.id.embeddable.SingleBoardComputerPk;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for the mappings of embeddable ids into MongoDB.
+ *
+ * @author Gunnar Morling
+ */
+public class EmbeddableIdMappingTest extends OgmTestCase {
+
+	@Before
+	public void setUpTestData() {
+		OgmSession session = openSession();
+		Transaction transaction = session.beginTransaction();
+
+		session.persist( new SingleBoardComputer( new SingleBoardComputerPk( "sbc-1" ), "Raspberry Pi" ) );
+
+		session.persist(
+				new News(
+						new NewsID( "seven-projects", "Jerry" ),
```

Comment:
It's a test for the specific persistent mapping, here we are asserting the
MongoDB-specific JSON representation.